astropy/coordinates/angles.py

```diff
@@ -650,7 +650,10 @@ def wrap_angle(self):
 
     @wrap_angle.setter
     def wrap_angle(self, value):
-        self._wrap_angle = Angle(value)
```

Comment:
I'm slightly bothered by this special-casing. How about adding `copy=False`? It is not quite as good, but I think in the overall initialization costs, it will be good enough:
```
In [3]: a = Angle(360., 'deg')

In [4]: %timeit Angle(a, copy=False)
The slowest run took 27.03 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 859 ns per loop

In [5]: %timeit Angle(a)
The slowest run took 15.84 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 9.1 µs per loop

In [6]: %timeit isinstance(a, Angle)
The slowest run took 34.33 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 72 ns per loop
```